packages/saltcorn-data/models/table.ts

```diff
@@ -152,24 +152,29 @@ class Table implements AbstractTable {
   /**
    *
    * Find one Table
+   *
    * @param where - where condition
    * @returns {*|Table|null} table or null
    */
-  static findOne(where: Where | Table): Table | null {
+  static findOne(where: Where | Table ): Table | null {
     if (
       where &&
       ((where.constructor && where.constructor.name === "Table") ||
         where.getRows)
     )
       return <Table>where;
+    // todo remove this logic. But needs find all usages! Also this is recursion that usually adds call function overhead
     if (typeof where === "string") return Table.findOne({ name: where });
     if (typeof where === "number") return Table.findOne({ id: where });
+    // todo i think here bug. External tables will be found only in one case (bottom if)
     if (where?.name) {
       const { getState } = require("../db/state");
       const extTable = getState().external_tables[where.name];
       if (extTable) return extTable;
     }
     const { getState } = require("../db/state");
+
+    // todo simplify  - I dont understand why this is too complex. And here potential bug because tables can be not cached!!!
```

Comment:
It's complex because it's a micro optimisation. It is calculating the anonymous function to pass to find, so we don't have to go through all of these conditionals for every element in the table array.